aspects/aspects.go

```diff
@@ -503,6 +498,60 @@ func (a *Aspect) Set(databag DataBag, request string, value interface{}) error {
 	return nil
 }
 
+func (a *Aspect) Unset(databag DataBag, request string) error {
+	if err := validateAspectDottedPath(request, nil); err != nil {
+		return badRequestErrorFrom(a, "unset", request, err.Error())
+	}
+
+	var matches []requestMatch
+	subkeys := strings.Split(request, ".")
+	for _, rule := range a.aspectRules {
+		placeholders, suffixParts, ok := rule.match(subkeys)
+		if !ok {
+			continue
+		}
+
+		if !rule.isWriteable() {
+			continue
+		}
+
+		path, err := rule.storagePath(placeholders)
+		if err != nil {
+			return err
+		}
+
+		matches = append(matches, requestMatch{
+			storagePath: path,
+			suffixParts: suffixParts,
+			request:     rule.originalRequest,
+		})
+	}
+
+	if len(matches) == 0 {
+		return notFoundErrorFrom(a, "unset", request, "no matching write rule")
+	}
```

Comment:
Is it worth putting this section into a function shared between Aspect.Set and Aspect.Unset?

Reply:
That's a good idea, thanks